src/vivarium_gates_nutrition_optimization/components/observers.py

```diff
@@ -1,6 +1,50 @@
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import PopulationView
 from vivarium_public_health.metrics.disease import DiseaseObserver
+from vivarium_public_health.utilities import to_years
+
+from vivarium_gates_nutrition_optimization.constants import models
 
 
 class PregnancyObserver(DiseaseObserver):
     def __init__(self):
         super().__init__("pregnancy")
+
+        # noinspection PyMethodMayBeStatic
+
+    def _get_population_view(self, builder: Builder) -> PopulationView:
+        columns_required = [
+            self.current_state_column_name,
+            self.previous_state_column_name,
+            "pregnancy_term_outcome",
+        ]
+        return builder.population.get_view(columns_required)
+
+    def on_collect_metrics(self, event: Event) -> None:
```

Comment:
This looks correct, but we will need to rewrite this observer entirely once VPH is released with the new results management system (hopefully in the next few days). If you'd like to get a headstart on the you can work off of an editable install of VPH on the branch `feature/results-manager-ng`.